pontoon/base/static/js/fluent_interface.js

```diff
@@ -198,25 +249,69 @@ var Pontoon = (function (my) {
       },
 
 
+      /*
+       * Is ast of a supported message?
+       *
+       * Message is supported if all value elements
+       * and all attribute elements are of type:
+       * - TextElement
+       * - ExternalArgument
+       * - MessageReference
+       * - SelectExpression?
+       */
+      isSupportedMessage: function (ast) {
+        function elementsSupported(elements) {
+          return elements.every(function(item) {
+            return [
+              'TextElement',
+              'ExternalArgument',
+              'MessageReference',
+              'SelectExpression'
+            ].indexOf(item.type) >= 0;
+          });
+        }
+
+        var valueSupported = !ast.value || elementsSupported(ast.value.elements);
+        var attributesSupported = ast.attributes.every(function (attribute) {
+          return attribute.value && elementsSupported(attribute.value.elements);
+        });
+
+        return valueSupported && attributesSupported;
+      },
+
+
+      /*
+       * Is element of type that can be presented as a simple string:
+       * - TextElement
+       * - ExternalArgument
+       * - MessageReference
+       */
+      isSimpleElement: function (element) {
+        return [
+          'TextElement',
+          'ExternalArgument',
+          'MessageReference'
+        ].indexOf(element.type) >= 0;
```

Comment:
Same here, I think that should be `=== 0`.

Reply:
Same as above.